### PR TITLE
Vim movement

### PR DESCRIPTION
**Pull request by robber-m**
Work from the weekend in Maine

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,6 +21,7 @@
 *.dylib
 
 # Executables
+ce
```
Comment (justy989) [inline .gitignore:24]: Oh man, this default gitignore file is lame, I think it's the default one that appears when you create a repo. I'll clean it up at some point so it only has the stuff we care about.

File: ce.c [hunk 2]
```diff
@@ -28,67 +31,76 @@ bool ce_alloc_lines(Buffer* buffer, int64_t line_count)
...
-          fread(contents, content_size, 1, file);
-          contents[content_size] = 0;
-
-          fclose(file);
-
-          buffer->filename = strdup(filename);
-     }
-
+void ce_load_string(Buffer* buffer, const char* str){
+     // TODO: merge with ce_insert_string
```
Comment (justy989) [inline ce.c:35]: When I do the merge, I'll see if I can get rid of ce_load_string() and just call ce_insert_string() in it's place

File: ce_config.c [hunk 3]
```diff
@@ -15,53 +26,116 @@ typedef struct{
      BufferChangeNode* changes_tail;
 } BufferState;
 
-bool initializer(BufferNode* head, Point* terminal_dimensions, int argc, char** argv, void** user_data)
+bool initialize_buffer(Buffer* buffer){
+     BufferState* buffer_state = malloc(sizeof(*buffer_state));
```
Comment (justy989) [inline ce_config.c]: calloc !

File: ce_config.c [hunk 4]
```diff
@@ -15,53 +26,116 @@ typedef struct{
...
+
+     return new_buffer_node;
+}
+
+BufferNode* new_buffer_from_file(BufferNode* head, const char* filename)
 {
-     // NOTE: need to set these in this module
-     g_message_buffer = head->buffer;
-     g_terminal_dimensions = terminal_dimensions;
+     Buffer* buffer = malloc(sizeof(*buffer));
```
Comment (justy989) [inline ce_config.c]: calloc !

File: ce_config.c [hunk 5]
```diff
@@ -159,35 +298,162 @@ bool key_handler(int key, BufferNode* head, void* user_data)
...
+               enter_insert_mode(config_state, cursor);
+          } break;
+          case 'c':
           case 'd':
-               // delete line
-               if(buffer->line_count){
-                    if(ce_remove_line(buffer, cursor->y)){
-                         if(cursor->y >= buffer->line_count){
-                              cursor->y = buffer->line_count - 1;
+               COMMAND{
```
Comment (justy989) [inline ce_config.c]: Not really liking the COMMAND macro, I think something like this is easier to read for me:

``` c
if(should_handle_vim_command(config_state, key)){
    // super cool stuff
}
```

Comment (justy989) [reply ce_config.c]: Hooray github markdown !

**Review by justy989 — COMMENTED** [review]
beep boop

File: .gitignore (hunk 1 above)
Comment (robber-m) [reply .gitignore:24]: Okay, works for me. It encompasses everything we're interested besides the main executable name so I'm fine with it for now

File: ce.c (hunk 2 above)
Comment (robber-m) [reply ce.c:35]: Okay, thanks

File: ce_config.c (hunk 3 above)
Comment (robber-m) [reply ce_config.c]: I'll change all of these. Thanks

File: ce_config.c (hunk 5 above)
Comment (robber-m) [reply ce_config.c]: huzzah. I'll do that :+1:  :smiley:  hooray github emoji's!

File: ce_config.c (hunk 4 above)
Comment (robber-m) [reply ce_config.c]: done

File: ce_config.c (hunk 3 above)
Comment (robber-m) [reply ce_config.c]: done

File: ce_config.c (hunk 5 above)
Comment (robber-m) [reply ce_config.c]: done

**Review by justy989 — APPROVED** [review]

**Pull request merged**